scripts/metrics-migration/migration.py

```diff
@@ -0,0 +1,640 @@
+
+                        # Write to pipe with chunking and gzip compression
+                        with ChunkedWriter(import_pipe_write, import_progress_bar) as chunked_writer:
+                            with gzip.GzipFile(fileobj=chunked_writer, mode="wb") as gzip_compressor:
+                                line_count = 0
+                                for transformed_bytes in transformer:
+                                    gzip_compressor.write(transformed_bytes)
+                                    line_count += 1
+                                
+                                export_progress_bar.set_postfix({"lines": line_count, "status": "exported"})
+            except Exception as e:
+                print(f"Error writing compressed data: {e}", file=sys.stderr)
+                raise
+            finally:
+                import_pipe_write.close()
+
+        # Start writer thread
+        writer_thread = threading.Thread(target=write_compressed_data, daemon=True)
+        writer_thread.start()
+
+        # Stream from pipe to import endpoint
+        try:
+            import_response = requests.post(
+                import_url,
+                data=import_pipe_read,
+                headers=import_headers,
+                auth=import_auth,
+                timeout=None,
+                verify=not SKIP_SSL_VERIFY,
+            )
+            import_response.raise_for_status()
+
+            import_progress_bar.set_postfix({"status": "imported"})
+        finally:
+            import_pipe_read.close()
+            writer_thread.join(timeout=300)  # Increased timeout for large migrations
+                
+    except requests.exceptions.RequestException as e:
+        print(f"Error during streaming migration: {e}", file=sys.stderr)
+        sys.exit(1)
+    except Exception as e:
+        print(f"Error during transformation: {e}", file=sys.stderr)
+        sys.exit(1)
+
+
+def import_metrics(
+    write_endpoint: str,
+    username: Optional[str],
+    password: Optional[str],
+    input_file: str,
+    progress_bar: tqdm,
+) -> None:
+    """
+    Import metrics to VictoriaMetrics API with streaming support.
+    
+    Args:
+        write_endpoint: VictoriaMetrics write endpoint
+        username: Authentication username (optional)
+        password: Authentication password (optional)
+        input_file: Path to input file (supports .gz extension)
+        progress_bar: Progress bar to update
+    """
+    url = f"{write_endpoint}".replace("/write", "/import")
+    headers = {}
+    auth = (username, password) if username and password else None
+
+    # Determine if input is gzipped
+    input_compress = input_file.endswith(".gz")
+    if input_compress:
+        headers["Content-Encoding"] = "gzip"
+
+    try:
+        # Open input file with appropriate mode
+        input_fp = open(input_file, "rb")
+
+        try:
+            # Get file size for progress tracking
+            input_fp.seek(0, 2)  # Seek to end
```

Comment:
Const is way better than magic number + comment:

```suggestion
            input_fp.seek(0, os.SEEK_END)
```